pandas/core/groupby/ops.py

```diff
@@ -732,119 +892,36 @@ def _cython_operation(
+        ngroups = self.ngroups
+
         func_uses_mask = cy_op.uses_mask()
         if is_extension_array_dtype(dtype):
             if isinstance(values, BaseMaskedArray) and func_uses_mask:
-                return self._masked_ea_wrap_cython_operation(
-                    cy_op, kind, values, how, axis, min_count, **kwargs
-                )
-            else:
-                return self._ea_wrap_cython_operation(
-                    cy_op, kind, values, how, axis, min_count, **kwargs
-                )
-
-        elif values.ndim == 1:
-            # expand to 2d, dispatch, then squeeze if appropriate
-            values2d = values[None, :]
-            res = self._cython_operation(
-                kind=kind,
-                values=values2d,
-                how=how,
-                axis=1,
-                min_count=min_count,
-                mask=mask,
-                **kwargs,
-            )
-            if res.shape[0] == 1:
-                return res[0]
-
-            # otherwise we have OHLC
-            return res.T
-
-        is_datetimelike = needs_i8_conversion(dtype)
-
-        if is_datetimelike:
-            values = values.view("int64")
-            is_numeric = True
-        elif is_bool_dtype(dtype):
-            values = values.astype("int64")
-        elif is_integer_dtype(dtype):
-            # e.g. uint8 -> uint64, int16 -> int64
-            dtype = dtype.kind + "8"
-            values = values.astype(dtype, copy=False)
-        elif is_numeric:
-            if not is_complex_dtype(dtype):
-                values = ensure_float64(values)
-
-        ngroups = self.ngroups
-        comp_ids, _, _ = self.group_info
-
-        assert axis == 1
-        values = values.T
-
-        if mask is not None:
-            mask = mask.reshape(values.shape, order="C")
-
-        out_shape = cy_op.get_output_shape(ngroups, values)
-        func, values = cy_op.get_cython_func_and_vals(values, is_numeric)
-        out_dtype = cy_op.get_out_dtype(values.dtype)
-
-        result = maybe_fill(np.empty(out_shape, dtype=out_dtype))
-        if kind == "aggregate":
-            counts = np.zeros(ngroups, dtype=np.int64)
-            if how in ["min", "max"]:
-                func(
-                    result,
-                    counts,
+                return cy_op._masked_ea_wrap_cython_operation(
                     values,
-                    comp_ids,
-                    min_count,
-                    is_datetimelike=is_datetimelike,
+                    min_count=min_count,
+                    ngroups=ngroups,
+                    comp_ids=comp_ids,
+                    **kwargs,
                 )
             else:
-                func(result, counts, values, comp_ids, min_count)
-        elif kind == "transform":
-            # TODO: min_count
-            if func_uses_mask:
-                func(
-                    result,
+                return cy_op._ea_wrap_cython_operation(
```

Comment:
Could we make `_ea_wrap_cython_operation` and some of the other private methods called from an instance of `WrappedCythonOp` public? Are there are any rules of thumb we have for private/public methods for non-user facing classes?

Reply:
good catch.  i think before long _cython_operation itself will move, at which point it will become public